libtransmission/session.cc

```diff
@@ -2520,6 +2522,20 @@ tr_port tr_sessionGetRPCPort(tr_session const* session)
     return tr_rpcGetPort(session->rpc_server_.get());
 }
 
+void tr_sessionSetRPCSocketMode(tr_session* session, int socket_mode)
+{
+    TR_ASSERT(tr_isSession(session));
+
+    tr_rpcSetRPCSocketMode(session->rpc_server_.get(), socket_mode);
+}
+
+int tr_sessionGetRPCSocketMode(tr_session const* session)
+{
+    TR_ASSERT(tr_isSession(session));
+
+    return tr_rpcGetRPCSocketMode(session->rpc_server_.get());
+}
+
```

Comment:
Are these functions needed? I don't think it makes sense to expose this feature in the GUI or in RPC (does it?), so we probably can make it exclusive to settings.json and remove the libtransmission public API

Reply:
It's up to you, I just mirrored similar public API functions. Since there's no mechanism to set the socket mode at runtime, they probably shouldn't be there.

Reply:
Let's omit it from this PR and then add it later if needed